shell/browser/api/electron_api_browser_window.cc

```diff
@@ -42,11 +42,23 @@ BrowserWindow::BrowserWindow(gin::Arguments* args,
   auto web_preferences = gin_helper::Dictionary::CreateEmpty(isolate);
   options.Get(options::kWebPreferences, &web_preferences);
 
+  bool transparent = false;
+  options.Get(options::kTransparent, &transparent);
+
+  std::string vibrancy_type;
+#if BUILDFLAG(IS_MAC)
+  options.Get(options::kVibrancyType, &vibrancy_type);
+#endif
+
   // Copy the backgroundColor to webContents.
   std::string color;
   if (options.Get(options::kBackgroundColor, &color)) {
     web_preferences.SetHidden(options::kBackgroundColor, color);
+#if BUILDFLAG(IS_WIN)
   } else if (window_->IsTranslucent()) {
+#else
+  } else if (!vibrancy_type.empty() || transparent) {
```

Comment:
This looks like what `NativeWindow::IsTranslucent()` checks for in its definition.

https://github.com/electron/electron/blob/3392d9a2e74973960ca516adc1c1684e03f78414/shell/browser/native_window.cc#L849-L860

Is that function not returning true for some reason? 🤔

Reply:
It is not, and I think it's because the transparency setting on the browser window and the native window is subtly different - trying to move this same check into the `IsTranslucent` method results in the same behavior we're seeing in main now (i.e. no vibrancy is set, it's just a white background)

Reply:
Ah, I think I see what's happening. `SetVibrancy` and `SetBackgroundMaterial` are getting called in `NativeWindow::InitFromOptions` which is called after we initialize the WebContents.

https://github.com/electron/electron/blob/fdf8b02e14a0f4a1fbbc77d741520ec4301a5ce3/shell/browser/native_window.cc#L254-L264

https://github.com/electron/electron/blob/f8dfc19bad06cd01f76d24414974b2a3687b0060/shell/browser/api/electron_api_browser_window.cc#L107-L108

It's unclear why we have two places that we're initializing options in NativeWindow, but if we move initializing the vibrancy and backgroundMaterial members to the constructor, it should fix the logic.

https://github.com/electron/electron/blob/fdf8b02e14a0f4a1fbbc77d741520ec4301a5ce3/shell/browser/native_window.cc#L95-L103

```diff
+#if BUILDFLAG(IS_WIN)
+  options.Get(options::kBackgroundMaterial, &background_material_);
+#elif BUILDFLAG(IS_MAC)
+  options.Get(options::kVibrancy, &vibrancy_);
+ #endif
```

Reply:
Oh great eye, you're right! Applied the changes, pushing up now 🙇‍♀️